src/components/GamePage.vue

```diff
@@ -0,0 +1,651 @@
+<script setup lang="ts">
+import { reactive } from 'vue';
+import { ref, onMounted, Vue} from "vue"
+
+const p = ref(null)
+const inputs = ref('') // Ref<string>型
+const message = (ele) => {
+  inputs += ele;
+}
+
+//キーボードのhashmap
+const keys: { [name: string]: string } = {};
+
+//1
+keys["1"] = "num1";
+keys["2"] = "num2";
+keys["3"] = "num3";
+keys["4"] = "num4";
+keys["5"] = "num5";
+keys["6"] = "num6";
+keys["7"] = "num7";
+keys["8"] = "num8";
+keys["9"] = "num9";
+keys["0"] = "num0";
+keys["-"] = "bar";
+keys["^"] = "hat";
+keys["¥"] = "¥";
+keys["Backspace"] = "Backspace";
+
+//2
+keys["q"] = "q";
+keys["w"] = "w";
+keys["e"] = "e";
+keys["r"] = "r";
+keys["t"] = "t";
+keys["y"] = "y";
+keys["u"] = "u";
+keys["i"] = "i";
+keys["o"] = "o";
+keys["p"] = "p";
+keys["@"] = "atmark";
+keys["["] = "bigparaini";
+keys["]"] = "bigparafini";
+
+//3
+keys["Control"] = "Control";
+keys["a"] = "a";
+keys["s"] = "s";
+keys["d"] = "d";
+keys["f"] = "f";
+keys["g"] = "g";
+keys["h"] = "h";
+keys["j"] = "j";
+keys["k"] = "k";
+keys["l"] = "l";
+keys[";"] = "semi";
+keys[":"] = "coron";
+keys["Enter"] = "Enter";
+
+//4
+keys["Shift"] = "Shift";
+keys["z"] = "z";
+keys["x"] = "x";
+keys["c"] = "c";
+keys["v"] = "v";
+keys["b"] = "b";
+keys["n"] = "n";
+keys["m"] = "m";
+keys[","] = "hai";
+keys["."] = "peri";
+keys["/"] = "slash";
+keys["_"] = "sub";
+
+//5
+keys["Alt"] = "Alt";
+keys["Meta"] = "Meta";
+keys[" "] = "space";
+
+//キーボード押したらひかる
+//ロジックとしては押したボタンのkey.valueを取得する。"."+key.valueというクラスリストを持つノードを探す。（それぞれに対応するクラスの名前はあらかじめひつ一つのclasslistに書いてある。そのノードに光る要素を加える。keyupしたらremove。
+const KeyDown = () => {
+  for(let key in keys){
+    if(key === event.key){
+      p.value.querySelectorAll("." + keys[key])[0].classList.remove("bg-gray-100")
+      p.value.querySelectorAll("." + keys[key])[0].classList.add("bg-indigo-500")
+    }
+  };
+};
+const KeyUp = () => {
+  for(let key in keys){
+    if(key === event.key){
+      p.value.querySelectorAll("." + keys[key])[0].classList.remove("bg-indigo-500")
```

Comment:
```suggestion
const KeyUp = (event) => {
  for(let key in keys){
    if(key === event.key){
      p.value.querySelectorAll("." + keys[key])[0].classList.remove("bg-indigo-500")
```
[悩み]
これもkeyDown()の指摘と同様です。